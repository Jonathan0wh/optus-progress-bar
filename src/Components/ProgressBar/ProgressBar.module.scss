.container {
  height: 50px;
  border: 1px solid lightgrey;
  margin: 16px;
  border-radius: 8px;
  position: relative;
}

.active-progress {
  height: 100%;
}

.label {
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  width: 100%;
  height: 100%;
}
