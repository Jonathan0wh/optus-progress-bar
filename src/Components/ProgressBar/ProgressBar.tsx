import styles from './ProgressBar.module.scss';

interface ProgressBarProps {
  currentPercentage?: number;
  testID?: string;
}

const ProgressBar = ({ currentPercentage = 0, testID }: ProgressBarProps) => {
  return (
    <div className={styles.container}>
      <div
        className={styles['active-progress']}
        style={{
          width: currentPercentage > 100 ? '100%' : currentPercentage + '%',
          backgroundColor: currentPercentage > 100 ? 'red' : 'lightblue',
          transition: ' width 1s',
        }}
      />
      <div className={styles.label} data-testid={testID}>
        {currentPercentage}%
      </div>
    </div>
  );
};

export default ProgressBar;
