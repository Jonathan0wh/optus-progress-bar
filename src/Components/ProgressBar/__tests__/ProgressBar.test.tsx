import { render, screen } from '@testing-library/react';

import ProgressBar from '../ProgressBar';

describe('ProgressBar Component', () => {
  test('render ProgressBar component with default 0% progress', () => {
    const { asFragment } = render(<ProgressBar />);
    const label = screen.getByText('0%');

    expect(label).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('render ProgressBar component with 50% progress', () => {
    const { asFragment } = render(<ProgressBar currentPercentage={50} />);
    const label = screen.getByText('50%');

    expect(label).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
