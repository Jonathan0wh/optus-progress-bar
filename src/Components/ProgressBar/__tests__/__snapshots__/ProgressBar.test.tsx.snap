// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component render ProgressBar component with 50% progress 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="active-progress"
      style="width: 50%; background-color: lightblue; transition: width 1s;"
    />
    <div
      class="label"
    >
      50%
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProgressBar Component render ProgressBar component with default 0% progress 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="active-progress"
      style="width: 0%; background-color: lightblue; transition: width 1s;"
    />
    <div
      class="label"
    >
      0%
    </div>
  </div>
</DocumentFragment>
`;
